lib/diyepw/diyepw/analyze_noaa_isd_lite_files.py

```diff
@@ -0,0 +1,140 @@
+       compression type (e.g. ".gz" for GZIP, ".zip" for ZIP, etc.) then you must pass this to indicate
+       what the files' compression format is. See the documentation of the `compression` parameter of
+       pandas.read_csv() for more information.
+
+    :return: dict in the form:
+    {
+      "good": [<file_description>, ...],
+      "too_many_total_rows_missing": [<file_description>, ...],
+      "too_many_consecutive_rows_missing": [<file_description>, ...]
+    }
+    ...where <file_description> is itself a dict in the form:
+    {
+      'file': <str> - absolute path to the file,
+      'total_rows_missing': <int> - The number of total rows that are missing data in this file,
+      'max_consec_rows_missing': <int> - The largest number of consecutive rows that are missing data in this file
+    }
+    """
+
+    too_many_missing_rows = []
+    too_many_consecutive_missing_rows = []
+    good_files = []
+
+    # Loop through all .gz files in the input directory - we filter on extension so that things like
+    # README files can be in that path without causing any problems. We filter on files that begin
+    # with 7 because all NOAA ISD Lite files for North America should start with 7, as all WMO indices
+    # in North America do.
+    for file in files:
+
+        # It's nicer to work with absolute paths, especially since we are going to put this path in a
+        # file to share with another script - otherwise that other script needs to know where this
+        # script is located to make sense of the relative paths
+        file = _os.path.abspath(file)
+
+        # Read the file into a Pandas dataframe.
+        df = _pd.read_csv(file,
+                         delim_whitespace=True,
+                         header=None,
+                         compression=compression,
+                         names=["Year", "Month", "Day", "Hour", "Air_Temperature",
+                                "Dew_Point_Temperature", "Sea_Level_Pressure", "Wind_Direction",
+                                "Wind_Speed_Rate", "Sky_Condition_Total_Coverage_Code",
+                                "Liquid_Precipitation_Depth_Dimension_1H", "Liquid_Precipitation_Depth_Dimension_6H"]
+                         )
+
+        # Take year-month-day-hour columns and convert to datetime stamps.
+        df['obs_timestamps'] = _pd.to_datetime(_pd.DataFrame({'year': df['Year'],
+                                                            'month': df['Month'],
+                                                            'day': df['Day'],
+                                                            'hour': df['Hour']}))
+
+        # Remove unnecessary year, month, day, hour columns
+        df = df.drop(columns=['Year', 'Month', 'Day', 'Hour'])
+
+        rows_present = df.shape[0]
+        file_description = {
+            'file': file,
+            'total_rows_missing': 8760 - rows_present,
```

Comment:
I don't think that's a good idea, though. The number of hours in a year isn't variable; it's either 8760 or it's 8784. We have a TODO to add handling for leap years so that we use the right one of those numbers based on the year represented by a file, but I'm having a hard time imagining a situation where we want to support the user passing in a custom number of hours in a year. That seems equivalent to letting a user define a custom number of inches in a foot or minutes in an hour.

Reply:
Then there needs to be a comment in the code that describes this if there is not one already.

Reply:
Added.